### PR TITLE
feat(module:carousel): add enhancement API and methods

### PR DESCRIPTION
**Pull request by HsuanXyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #652 

## What is the new behavior?

Added the following API & methods

#### API

| 参数 | 说明 | 类型 | 默认值 |
| --- | -- | -- | -- |
| nzAutoPlaySpeed | 自动切换速度, 单位毫秒 | Number | 3000 |
| nzPauseOnHover | 鼠标悬停暂停 | Boolean | true |
| nzAfterChange | 切换面板后的回调, 参数为当前索引 | Function(number) | 无 |
| nzBeforeChange | 切换面板前的回调, 参数为当前的索引与被切换到的索引 | Function({ form: number; to: number}) | 无 |

#### Methods

| 方法名 | 参数 | 说明 | 返回值值 |
| --- | -- | -- | -- |
| nzSlickNext | 无 | 切换到下一个面板 | 无 |
| nzSlickPrev | 无 | 切换到下一个面板 | 无 |
| nzSlickGoTo | index: number | 切换到指定索引面板 | 无 |

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by vthinkxie** [comment]
Hi @HsuanXyz , please rebase master and make all boolean `@Input` support attributes usage as https://github.com/NG-ZORRO/ng-zorro-antd/pull/459/files#diff-26c9eaa42ed9a7448f1a559c238529d3R68

**Comment by HsuanXyz** [comment]
@vthinkxie OK now.

File: src/components/carousel/nz-carousel.component.ts [hunk 1]
```diff
@@ -88,9 +138,11 @@ export class NzCarouselComponent implements AfterViewInit, OnDestroy {
   }
 
   setActive(content, i) {
-    if (this.nzAutoPlay) {
+    if ((this.nzAutoPlay && !this.nzPauseOnHover) || (this.nzAutoPlay && !this._mouseHover)) {
```
Comment (vthinkxie) [inline src/components/carousel/nz-carousel.component.ts]: should be
```typescript
if ((this.nzAutoPlay && !this.nzPauseOnHover) || (this.nzAutoPlay && this.nzPauseOnHover && !this._mouseHover)){
}
```

**Comment by vthinkxie** [comment]
@HsuanXyz

**Comment by HsuanXyz** [comment]
@vthinkxie Sorry I botch up the git log, new PR #688

**Pull request closed without merging**